crates/unleash-edge-http-client/src/instance_data.rs

```diff
@@ -93,48 +159,29 @@ pub fn create_send_instance_data_task(
                 std::time::Duration::from_secs(60) + delay * std::cmp::min(errors, 10),
             )
             .await;
-            match instance_data_sender.as_ref() {
-                InstanceDataSending::SendNothing => {
-                    debug!("No instance data sender found. Doing nothing.");
-                    continue;
+
+            let result = send_instance_data(
+                &instance_data_sender,
+                &our_instance_data,
+                &downstream_instance_data,
+            )
+            .await;
+            match result {
+                Ok(_) => {
+                    debug!("Successfully posted observability metrics.");
+                    errors = 0;
+                    downstream_instance_data.write().await.clear();
+                    our_instance_data.clear_time_windowed_metrics();
                 }
-                InstanceDataSending::SendInstanceData(instance_data_sender) => {
-                    let status = send_instance_data(
-                        instance_data_sender,
-                        our_instance_data.clone(),
-                        downstream_instance_data.clone(),
-                    )
-                    .await;
-                    if let Err(e) = status {
-                        match e {
-                            EdgeError::EdgeMetricsRequestError(status, _) => {
-                                if status == StatusCode::NOT_FOUND {
-                                    debug!(
-                                        "Our upstream is not running a version that supports edge metrics."
-                                    );
-                                    errors += 1;
-                                    downstream_instance_data.write().await.clear();
-                                    our_instance_data.clear_time_windowed_metrics();
-                                } else if status == StatusCode::FORBIDDEN {
-                                    warn!(
-                                        "Upstream edge metrics said our token wasn't allowed to post data"
-                                    );
-                                    errors += 1;
-                                    downstream_instance_data.write().await.clear();
-                                    our_instance_data.clear_time_windowed_metrics();
-                                }
-                            }
-                            _ => {
-                                warn!("Failed to post instance data due to unknown error {e:?}");
-                            }
-                        }
-                    } else {
-                        debug!("Successfully posted observability metrics.");
-                        errors = 0;
-                        downstream_instance_data.write().await.clear();
-                        our_instance_data.clear_time_windowed_metrics();
+                Err(err) => match err {
```

Comment:
This code can now get a nice error which it can use to determine whether or not it should backoff, allow the backoff state to persist or clear the backoff state without having to deal with the http error messages